cpp/oneapi/dal/backend/primitives/placement/argwhere.hpp

```diff
@@ -0,0 +1,222 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include <tuple>
+#include <limits>
+
+#include "oneapi/dal/detail/profiler.hpp"
+
+#include "oneapi/dal/backend/primitives/common.hpp"
+#include "oneapi/dal/backend/primitives/ndarray.hpp"
+#include "oneapi/dal/backend/primitives/reduction.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+enum class where_alignment : std::int64_t { left = 0b0l, right = 0b1l };
+
+#ifdef ONEDAL_DATA_PARALLEL
+
+template <typename Index, where_alignment align>
+struct where_alignment_map {};
+
+template <typename Index>
+struct where_alignment_map<Index, where_alignment::left> {
+    constexpr static inline sycl::ext::oneapi::maximum<Index> binary{};
+    constexpr static inline auto identity = std::numeric_limits<Index>::lowest();
+};
+
+template <typename Index>
+struct where_alignment_map<Index, where_alignment::right> {
+    constexpr static inline sycl::ext::oneapi::minimum<Index> binary{};
+    constexpr static inline auto identity = std::numeric_limits<Index>::max();
+};
+
+template <typename Functor,
+          typename Type,
+          typename Index,
+          where_alignment align = where_alignment::right>
+inline sycl::event argwhere_one(sycl::queue& queue,
```

Comment:
Really need a description to those. It seems it does something like numpy's `argwhere`, but why is there `_one`?

Reply:
No. Numpy can return multiple values as a result. That's why it have this postfix

![image](https://github.com/oneapi-src/oneDAL/assets/60149822/b4c253f0-112e-46af-9fd9-5950f340aefa)